### ISSUE TITLE
Cannot dump definition because of invalid class name

### ISSUE THREAD

**Issue by osr7** [symfony/symfony#23011]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | yes
| RFC?             | no
| Symfony version  | 3.3.0

After upgrading from 3.2.9 to 3.3.0 I get an error after using `matejvelikonja/LabbyBundle`.
There seems to be an regression in 3.3.0

```
 [RuntimeException]                                                               
  An error occurred when executing the "'cache:clear --no-warmup'" command:        
  PHP Fatal error:  Uncaught Symfony\Component\DependencyInjection\Exception\Runt  
  imeException: Cannot dump definition because of invalid class name ('velikonja_  
  labby.pre_sync') in /vagrant/vendor/symfony/symfony/src/Symfony/Component/Depen  
  dencyInjection/Dumper/PhpDumper.php:1600                                         
  Stack trace:                                                                     
  #0 /vagrant/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Du  
  mper/PhpDumper.php(1471): Symfony\Component\DependencyInjection\Dumper\PhpDumpe  
  r->dumpLiteralClass(''velikonja_labb...')                                        
  #1 /vagrant/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Du  
  mper/PhpDumper.php(479): Symfony\Component\DependencyInjection\Dumper\PhpDumper  
  ->dumpValue(Object(Symfony\Component\DependencyInjection\Argument\ClosureProxyA  
  rgument))                                                                        
  #2 /vagrant/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Du  
  mper/PhpDumper.php(695): Symfony\Component\DependencyInjection\Dumper\PhpDumper  
  ->addServiceMethodCalls('debug.event_dis...', Object(Symfony\Component\Dependen  
  cyInjection\Definition))                                                         
  #3 /vagrant/vendor/symfony/symfony/src/Sym in /vagrant/vendor/symfony/symfony/s  
  rc/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php on line 1600       
   ```                                                 
https://github.com/matejvelikonja/LabbyBundle/issues/15

**Comment by weaverryan** [comment]
There are 2 things going on here:

1) The actual problem is that this bundle contains an event subscriber that is using undefined method. So, `velikonja_labby.pre_sync` is actually the name of the method the subscriber is attached to. The error from `PhpDumper` is misleading: it's an invalid *method* name, not an invalid *class* name.

**This error could be improved**

2) As crazy as it sounds to subscribe using a method whose name is `velikonja_labby.pre_sync`, it *was* technically legal. The bundle uses a magic `_call` method (https://github.com/matejvelikonja/LabbyBundle/blob/f947657acf401eb605498e407e3c6e59c3a51437/Executor/ExecutorRunner.php#L34) to dispatch the "fake method name" to a real method.

**Can/should we fix this?**

**Comment by nicolas-grekas** [comment]
Yes we can, we should just not call dumpLiteralClass on L1471, this looks superflous (not done in other places when dumping methods, isn't it?)

**Comment by chalasr** [comment]
Is this fixed by #23008?

**Comment by nicolas-grekas** [comment]
yes :)
